vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting.cc

```diff
@@ -140,10 +141,18 @@ void SubdivisionTargeting::MaybeAllowForLocale(const std::string& locale) {
   }
 
   const std::string country_code = brave_l10n::GetCountryCode(locale);
-  const SupportedSubdivisionCodesSet& subdivision_codes =
-      kSupportedSubdivisionCodes.at(country_code);
+  const std::string country_code_part = country_code + "-";
 
   const std::string& subdivision_code = GetSubdivisionCode();
+  if (!base::StartsWith(subdivision_code, country_code_part)) {
```

Comment:
Can we not use `vendor/bat-native-ads/src/bat/ads/internal/base/locale/subdivision_code_util.h` to parse out the country code and subdivision codes?